### ISSUE TITLE
The `data_time` is much bigger in the 1.x version

### ISSUE THREAD

**Issue by leexinhao** [open-mmlab/mmaction2#2200]
Thanks for your error report and we appreciate it a lot.
If you feel we have help you, give us a STAR! :satisfied:

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**

I try to use the 1.x version in my work, but I found that the `data_time` is much bigger in the 1.x version than 0.x version when the same configuration is used：

![image](https://user-images.githubusercontent.com/48858574/216773895-8fe3c489-e8fa-474c-84cc-e2902dd7381b.png)

![image](https://user-images.githubusercontent.com/48858574/216774007-4cec8b52-3242-460e-b2b2-f31bca79b66b.png)

As you can see, **the `data_time`  of 1.x version is very big in the first iteration, and in later iterations it is still ten times larger than the 0.x version**, when I increase the gpu_nums from 1 to 4, things are getting worse!!!, we need **42.8981 time** to prepare the data of fisrt iter:

![image](https://user-images.githubusercontent.com/48858574/216774258-807e2918-f542-4af1-8193-a657b8590e99.png)

**Reproduction**

1. What command or script did you run?

For mmaction2 1.x：
```
bash tools/dist_train.sh  configs/recognition/tsn/tsn_imagenet-pretrained-r50_8xb32-1x1x8-100e_kinetics400-rgb.py 1
```
For mmaction2 0.x:
```
bash tools/dist_train.sh configs/recognition/tsn/tsn_r50_1x1x8_100e_kinetics400_rgb.py 1
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?

I just modify the data_root, and I make sure that it's same in both two configs file.

3. What dataset did you use?
```python
# dataset settings
dataset_type = 'VideoDataset'
data_prefix = 'lxh:s3://lxhBucket/kinetics_400_320_30fps/'
data_root = data_prefix+'kinetics_400_320_30fps_train'
data_root_val = data_prefix+'kinetics_400_320_30fps_val'
ann_file_train = 'data/kinetics400/kinetics400_train_list_videos.txt'
ann_file_val = 'data/kinetics400/kinetics400_val_list_videos.txt'
ann_file_test = 'data/kinetics400/kinetics400_val_list_videos.txt'
```
**Environment**

1. Please run `PYTHONPATH=${PWD}:$PYTHONPATH python mmaction/utils/collect_env.py` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
   - How you installed PyTorch \[e.g., pip, conda, source\]
   - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)
For mmaction2 1.x:
```
sys.platform: linux
Python: 3.8.0 (default, Nov  6 2019, 21:49:08) [GCC 7.3.0]
CUDA available: False
numpy_random_seed: 2147483648
GCC: gcc (GCC) 5.4.0
PyTorch: 1.8.1+cu101
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.1, CUDNN_VERSION=7.6.3, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.9.1+cu101
OpenCV: 4.7.0
MMEngine: 0.5.0
MMAction2: 1.0.0rc2+8f96223
```
For mmaction2 0.x:
```
sys.platform: linux
Python: 3.7.13 (default, Mar 29 2022, 02:18:16) [GCC 7.5.0]
CUDA available: False
GCC: gcc (GCC) 5.4.0
PyTorch: 1.8.1+cu101
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.1, CUDNN_VERSION=7.6.3, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.9.1+cu101
OpenCV: 4.7.0
MMCV: 1.7.0
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMAction2: 0.24.1+09426f5
```
**Error traceback**

If applicable, paste the error traceback here.

```
A placeholder for traceback.
```

**Bug fix**

I don't know, maybe it was because of the `mmengine`，but I never encounter this issue in `mmclassification1.0` which also uses the `mmengine`.

**Comment by cir7** [comment]
Hi, @leexinhao . I tried to reproduce your problem but failed. In my environment, the `datatime` in branch 0.x and 1.x has no significant difference for the first iteration. And I found that the default `num_workers` is different in two branches, how about trying to adjust it?
Another point is that TSN training with a larger batchsize, compared to other models, may more likely cause a `datatime` issue. How about try compare other models in two branches?

**Comment by leexinhao** [comment]
> 

I set `num_workers` to 8 for two configs, and I tried smaller values and came to similar conclusions. By the way, I use the ceph (object-based storage) of my company:
```python
file_client_args = dict(io_backend='petrel')
```
Is it possible that mmengine and the original mmcv treated this piece differently so that I meet this issue.

TSM do use a larger batchsize, but I think there should be no significant difference between the two branches using any configs, which is where I get confused. I have tried this command to train a timesformer:

```python
bash tools/dist_train.sh configs/recognition/timesformer/timesformer_divST_8xb8-8x32x1-15e_kinetics400-rgb.py 1 
```

And it's also very slow:
![image](https://user-images.githubusercontent.com/48858574/216979173-6fc28554-edcf-47ea-92d4-3461df60f9b0.png)


**What the version of your mmengine?** Another issue I encounter is need to wait a long time (**about 10~20 minutes**) before the train/test progress bar appear in the terminal. I had the same problem when using the mmclassification, so I thought maybe my version of mmengine has somthing wrong.

**Comment by cir7** [comment]
@leexinhao 
Actually, we are using the same backend(petrel), and I upgrade mmengine to 0.5.0 as you.

**Comment by leexinhao** [comment]
> @leexinhao Actually, we are using the same backend(petrel), and I upgrade mmengine to 0.5.0 as you. Do you mind giving me your email, so that we can communicate more conveniently?

ok，xinhaoli00@outlook.com

**Comment by leexinhao** [comment]
@cir7 

And I change to another machine with new environment which also encounter this problem:
![image](https://user-images.githubusercontent.com/48858574/217419402-fd1f9529-334e-48c4-95e6-c556d71590ae.png)

**Comment by cir7** [comment]
The issue is caused by an incompatible of remote data backend SDK, we fixed it by updating the SDK.

**Comment by bc-arl** [comment]
Hi, I am also getting this issue, same setup as leexinhao described, but loading from disk. (Data is stored on a hard drive plugged into computer). I am only finetuning the final layer of the model so there is no reason for the training to take so long, I think it is a dataloader issue. Can you explain how to fix it in more detail (i.e how do I "update remote data backend sdk"?)
Thanks for the help!
`
06/14 09:29:29 - mmengine - INFO - Epoch(train)  [1][ 20/919]  lr: 5.0000e-03  eta: 11 days, 11:28:59  time: 21.5924  data_time: 21.4400  memory: 319  grad_norm: 17.2139  loss: 3.5321  top1_acc: 0.0000  top5_acc: 0.5000  loss_cls: 3.5321
06/14 09:37:35 - mmengine - INFO - Epoch(train)  [1][ 40/919]  lr: 5.0000e-03  eta: 12 days, 4:33:55  time: 24.2902  data_time: 24.2473  memory: 319  grad_norm: 15.5173  loss: 3.2071  top1_acc: 0.0000  top5_acc: 0.5000  loss_cls: 3.2071
`
